### ISSUE TITLE
anything to do with packages is very slow

### ISSUE THREAD

**Issue by aghaynes** [rstudio/rstudio#4191]
### System details

    RStudio Edition : Desktop
    RStudio Version : 1.2.1237
    OS Version      :  Windows 10
    R Version       : 3.5.1

### Steps to reproduce the problem
No idea

### Describe the problem in detail
Anything to do with packages is very slow (loading, installing, compiling the list of installed packages in the RStudio pane, ...). This issue appeared a few weeks ago (as far as I can tell, no particular changes happened to create the issue, at least not at the users end). I vaguely remember a similar issue on an old discussion thread where it was suggested that the problem might be due to network settings...
My computer is part of a hospital network where our desktops are available from any computer in the hospital (R and RStudio need to be installed individually though). Another R user has mentioned that the problem is abated somewhat on WLAN (as opposed to LAN) and almost goes away when disconnected from the network (personally, I cannot test this, as I have a desktop that anyway connects via WLAN). 

I attach the diagnostics report...

[diagnostics-report.txt](https://github.com/rstudio/rstudio/files/2782948/diagnostics-report.txt)

### Describe the behavior you expected
fast loading/installing as per on other systems (there is virtually no lag on other computers I have used RStudio on)

**Comment by ronblum** [comment]
@aghaynes Thank you for raising the issue. Does the slowness occur if you run this in R, outside of RStudio?

**Comment by kevinushey** [comment]
This community post might be relevant: https://community.rstudio.com/t/rstudio-suddenly-slow-processing/4959/18

**Comment by aghaynes** [comment]
It is also slow in R, but not as slow (that's my impression at least), although I've not timed it...I can try that.

@kevinushey's point with the home variable, I had also wondered about this but wasn't sure what it was referring to (I noticed that there was an error in the diagnostic report about that). Unfortunately, as it's a hospital computer with extremely strict computer policies, I doubt that I'll be able to change it... (I now see that it's the HOME environment variable)

**Comment by kevinushey** [comment]
One escape hatch that might help: you can disable the Packages pane entirely, which disables much of the package-related background tasks performed by RStudio.

<img width="599" alt="screen shot 2019-01-23 at 8 34 12 am" src="https://user-images.githubusercontent.com/1976582/51621578-b1c1eb80-1ee9-11e9-8a8f-c0e64b57c5e3.png">

You could try unchecking the `[x] Enable Packages pane` checkbox and restarting RStudio.

**Comment by r1ruest** [comment]
Hi together
I'm working with Alan and I have the same problem. I hope I can help you to solve it.
@ronblum I compared the time for doing this in RStudio and R:
Sys.time()
installed.packages()
Sys.time()
I attached the results. It did take a long time in R and RStudio. I tested it twice, right one after the other. The first time it took longer than a minute, but the second time only 1 second. The same for R.

[RStudio.txt](https://github.com/rstudio/rstudio/files/2792253/RStudio.txt)
[R.txt](https://github.com/rstudio/rstudio/files/2792254/R.txt)

@kevinushey unchecking the [x] Enable Packages pane did not change anything in the speed.

Not only loading a package is very slow, also just opening R or RStudio takes over a minute.
Also setting the working director using tab (indent key) to see a selection for the path is very slow.

What I realized by accident is, at some time, I had no connection to one of the network-folders (I don't know why), the speed was normal. After having the connection again to to this network it was slow again. This correspond with the same result I have when I use either "WLAN" nor "LAN". No problem when I'm disconnected.

**Comment by ronblum** [comment]
@r1ruest In line with the RStudio Community post @kevinushey referenced, I tried both with a (local) network drive in UNC format and with mapping it to a letter driver, and I noticed a small delay with the UNC patch vs. mapped drive when running `installed.packages`. So you might try that—it's worked in the past (as in the post) for remote drive issues.

If I'm reading the results correctly, it was about 3.5 minutes in RStudio and 2.5 minutes in R. 
So this looks like an issue with R. This forum is meant for issues with RStudio, so you might consider posting to the RStudio Community](https://community.rstudio.com/) post that's listed above or creating a new one. Another option is [Stack Overflow](https://stackoverflow.com/questions/tagged/r), where other R users can provide help.
